libs/designer-ui/src/lib/floatingactionmenu/floatingactionmenubase/index.tsx

```diff
@@ -0,0 +1,147 @@
+    const closeErrorButtonAriaLabel = intl.formatMessage({
+      defaultMessage: 'Close',
+      description: 'Close button aria label',
+    });
+
+    return (
+      <div>
+        <span className="msla-floating-action-menu-items-title">{props.expandedTitle}</span>
+        <span
+          role="button"
+          aria-label={closeErrorButtonAriaLabel}
+          className="msla-floating-action-menu-items-close"
+          onClick={toggleExpanded}
+          onKeyDown={toggleExpandedOnKeyDown}
+          tabIndex={0}
+        >
+          {'x'}
+        </span>
+      </div>
+    );
+  };
+
+  const renderMenuItem = (menuItem: FloatingActionMenuItem): JSX.Element => {
+    const itemStyle = {
+      background: `url('${menuItem.icon}') no-repeat center`,
+    };
+
+    const handleMenuItemSelected = (item: FloatingActionMenuItem): void => {
+      toggleExpanded();
+
+      props.onMenuItemSelected(item);
+    };
+
+    const handleMenuItemSelectedOnKeyDown = (e: React.KeyboardEvent<HTMLElement>, item: FloatingActionMenuItem): void => {
+      const { keyCode } = e;
+
+      if (keyCode === KeyCodes.enter || keyCode === KeyCodes.space) {
+        e.preventDefault();
+        e.stopPropagation();
+        handleMenuItemSelected(item);
+      }
+    };
+
+    if (menuItems.length === 1) {
+      return (
+        <div
+          role="button"
+          aria-label={menuItem.label}
+          tabIndex={0}
+          className="msla-floating-action-menu-item-vertical-container"
+          onClick={() => handleMenuItemSelected(menuItem)}
+          onKeyDown={(e: React.KeyboardEvent<HTMLElement>): void => handleMenuItemSelectedOnKeyDown(e, menuItem)}
+        >
+          <div className="msla-menu-item-logo" style={itemStyle} />
+          <span className="msla-vertical-menu-item-label">{menuItem.label}</span>
+        </div>
+      );
+    } else {
+      return (
+        <div key={menuItem.type} className="msla-floating-action-menu-item-horizontal-container">
+          <div
+            role="button"
+            aria-label={menuItem.label}
+            tabIndex={0}
+            className="msla-menu-item-logo"
+            style={itemStyle}
+            onClick={() => handleMenuItemSelected(menuItem)}
+            onKeyDown={(e: React.KeyboardEvent<HTMLElement>): void => handleMenuItemSelectedOnKeyDown(e, menuItem)}
+          />
+          <span className="msla-horizontal-menu-item-label">{menuItem.label}</span>
+        </div>
+      );
+    }
+  };
+
+  const renderMenuItems = (): JSX.Element => {
+    return (
+      <div className="msla-floating-action-menu-items-container">
+        {renderMenuItemsHeader()}
+        <div className="msla-floating-action-menu-items">
+          {menuItems.map((item: FloatingActionMenuItem) => {
+            return renderMenuItem(item);
+          })}
+        </div>
+      </div>
+    );
+  };
+
+  return (
+    <>
+      <div className="msla-dynamic-added-params-container">{props.children}</div>
```

Comment:
This file is pretty much all the UI code (the main similarity btw. FloatingActionMenuInputs and Outputs)